CodeEdit/Features/WindowCommands/FileCommands.swift

```diff
@@ -44,12 +44,16 @@ struct FileCommands: Commands {
             .keyboardShortcut("w", modifiers: [.control, .shift, .command])
 
             Button("Close Window") {
-
+                NSApp.keyWindow?.close()
             }
             .keyboardShortcut("w", modifiers: [.shift, .command])
 
             Button("Close Workspace") {
-
+                NSDocumentController().closeAllDocuments(
```

Comment:
You're creating a new, unused, NSDocumentController here. What you probably want to do it use `NSDocumentController.shared.closeAllDocuments`